### ISSUE TITLE
figure 3 clarification from publication

### ISSUE THREAD

**Issue by jessielyons** [aspuru-guzik-group/Pasithea#6]
Dear Authors,

Very nice publication.

1) Can you please clarify how you plot the figure 3? Is training loss vs epoch plotted from dreaming or training. Since you mentioned target logp in figure as 3 or -3, i am guessing its from dreaming step.

2) Can you please also clarify what is the meaning of logP vs epochs in figure 3? Is it from dreaming step? Can you share what are plotting from outputs generated? Are you using log file of the run or using "sampled_intermediate_mol" file for a particular target logp?

Any related code on how to generate Figure 3 can help.

Thanks so much,
JL

**Comment by MarioKrenn6240** [comment]
ad1) yes, loss is from the dreaming process

ad2) logP is computed via RDKit (its for testing the ground truth of our system, and thats why its not a smooth function because we use onehot). Loss is the evolution of the loss during dreaming, which is the MSE between model prediction and target property.

The code for this is basically written in demo.py, let me know if this helps.

**Comment by jessielyons** [comment]
Hey Mario,

Is it possible for you to provide me with a modified demo.py without use of
"lst_of_logP" anywhere in dreaming (def dream and def dream_model). This
will mean that we don't have any ground truth information available? This
way I can guarantee to just replace wherever logP is used in your code with
my target property of interest. I don't have a ground truth for dreamed
molecules based on training on my input smiles and associated experimental
property. I changed input everywhere else but got confused when I reached
the dreaming step.

Apologies, it can get complicated trying to modify the code myself and
remove all the code associated with utilizing the ground truth logP which
is the calculated logP.

Best,
Jessie

**Comment by MarioKrenn6240** [comment]
Sure, you can just delete these lines, and it will work straightforwardly. The main idea is between line 283 and 300, so just go ahead and remove the rest and adjust the output of the dream function.

**Comment by jessielyons** [comment]
Awesome, thank you so much for all your quick and helpful replies! I will
try accordingly.

Best,
JL
